server/polar/customer_portal/endpoints/subscription.py

```diff
@@ -104,6 +105,72 @@ async def get(
     return subscription
 
 
+class SubscriptionChargePreviewResponse(TypedDict):
+    base_amount: int
+    subtotal_amount: int
+    tax_amount: int
+    total_amount: int
+
+
+@router.get(
+    "/{id}/charge-preview",
+    summary="Preview Next Charge For Active Subscription",
+    response_model=SubscriptionChargePreviewResponse,
+    responses={404: SubscriptionNotFound},
+)
+async def get_charge_preview(
+    id: SubscriptionID,
+    auth_subject: auth.CustomerPortalRead,
+    session: AsyncSession = Depends(get_db_session),
+) -> SubscriptionChargePreviewResponse:
+    """Get current period usage and cost breakdown for a subscription."""
+    subscription = await customer_subscription_service.get_by_id(
+        session, auth_subject, id
+    )
+
+    if subscription is None:
+        raise ResourceNotFound()
+
+    if subscription.status != "active":
+        ## FIXME Is a 404 the correct behavior?
+        raise ResourceNotFound()
+
+    base_price = next(p for p in subscription.prices if p.amount_type == "fixed")
+
+    subtotal = base_price.price_amount + sum(
+        meter.amount for meter in subscription.meters
+    )
+
+    tax_amount = 0
+
+    if (
+        subscription.product.is_tax_applicable
+        and subscription.customer.billing_address is not None
+    ):
+        tax = await calculate_tax(
+            subscription.id,
+            subscription.currency,
+            subtotal,
+            subscription.product.stripe_product_id,
+            subscription.customer.billing_address,
+            [subscription.customer.tax_id]
+            if subscription.customer.tax_id is not None
+            else [],
+            subscription.tax_exempted,
+        )
+
+        tax_amount = tax["amount"]
+
+    total = subtotal + tax_amount
```

Comment:
We should also probably consider the discount. The tricky part is that we'll need to check if the discount is still applicable, like we do in `cycle`.

https://github.com/polarsource/polar/blob/e0e7b3339b6703969c4a24c8049f325c97d81dc5/server/polar/subscription/service.py#L590-L596

https://github.com/polarsource/polar/blob/e0e7b3339b6703969c4a24c8049f325c97d81dc5/server/polar/subscription/service.py#L613-L617